release/models/platform/openconfig-platform-chassis.yang

```diff
@@ -0,0 +1,79 @@
+module openconfig-platform-chassis {
+  yang-version "1";
+
+  namespace "http://openconfig.net/yang/platform/chassis";
+
+  prefix "oc-platform-chassis";
+
+  import openconfig-platform { prefix oc-platform; }
+  import openconfig-extensions { prefix oc-ext; }
+
+  organization "OpenConfig working group";
+  contact
+    "OpenConfig working group
+    www.openconfig.net";
+
+  description
+    "This module defines extensions to the OpenConfig platform model
+    that apply to chassis (CHASSIS) components.
+    These components are global to the device hardware package and
+    are not already part of another platform module and which
+    configuration or state is applicable.";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision "2022-04-22" {
+    description
+      "Initial revision with chassis-id.";
+    reference "0.1.0";
+  }
+
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+
+  // extension statements
+
+  // feature statements
+
+  // identity statements
+
+  // typedef statements
+
+  // grouping statements
+
+  grouping chassis-config {
+    description
+      "Config that corresponds to the global hardware package";
+
+    leaf chassis-id {
```

Comment:
Is there a use of this outside of P4RT? If there isn't, then I suggest that we should specify it (like we do with `node-id`) in the `openconfig-p4rt` module rather than here. [reference](https://github.com/openconfig/public/blob/master/release/models/p4rt/openconfig-p4rt.yang)

I think this makes it clearer what this is used for, and gives more context vs. being a generic "ID".

Reply:
Thanks for the suggestion, Rob! TBH, even for `node-id`, it is used beyond openconfig-p4rt in WBB and gPINS. For each singleton port, we associate it with a `node-id`, so that it could be processed by corresponding SFE node (chip node) when there is any update about the singleton port. Then the update event will be passed to upper-layer applications (e.g., TM). 

For example, eth1 is controller by chip1 in the physical switch. Then the port status of eth1 turns to UP, this info will be sensed by gNMI client (as we subscribe to interface state path). Based on switch hierarchy (interface->port->chip), the controller knows that eth1 belongs to chip1 (node-id is 999), and distributes this port state update along with other info (i.e., a high-level representation) to SFE chip1 node, based on the `node-id`. Then the SFE chip1 node is responsible for dispatching the high-level representation to upper layer applications. 

The chassis-id serves the same purpose here, but it is used for chassis node. Since p4rt does not have a sense of chassis, it might not be appropriate to move under `openconfig-p4rt`. But I do not have strong objection for moving if we want to keep consistence.

Reply:
I think my preference for this to be somewhere other than just in the base platform model is that this is very much a SDN concept, I'm not sure that I've seen it be used in other approaches. Even if we take @dplore's approach with a configurable ID I think we should probably put it in some form of extension module.

Reply:
Moved this to the p4rt module as it is relevant to the use case